src/btree/bt_curnext.c

```diff
@@ -402,6 +403,17 @@ __cursor_row_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart, size_t *skip
 restart_read_page:
         rip = &page->pg_row[cbt->slot];
         WT_RET(__cursor_row_slot_key_return(cbt, rip, &kpack, &kpack_used));
+        /*
+         * If the cursor has prefix search configured we can early exit here if the key that we are
+         * visiting is after our prefix.
+         */
+        if (F_ISSET(&cbt->iface, WT_CURSTD_PREFIX_SEARCH) && prefix != NULL &&
+          __wt_prefix_match(prefix, &cbt->iface.key) == -1) {
```

Comment:
I think our __wt_lex_compare reserves the right to return something other than -1, 0, 1. Could you check `< 0` here instead? Similarly for the check in `cursor_row_prev`?

Reply:
Fixed now 👍